synonymdev/bitkit-react-native
# feat(lightning): Implemented Custom Channels

This PR:
 - Adds the ability to purchase Blocktank channels via the Custom Channels flow.

To Test:
1. Ensure the app has around at least 20,000-100,000 sats.
2. Select the "Get on Lightning" suggestion on the Home screen or navigate to:
   - Settings->Networks->Lightning Connections->Add New Connection
3. Select "Custom Setup".
4. Keep "0" selected as your spending balance and press "Continue".
5. Keep the pre-set receiving amount and press "Continue".
6. Swipe to pay on the confirm screen.
7. Once confirmed, mine a block or two and wait for Blocktank to see the transaction and open the channel.
    - If the channel doesn't open after some time attempt to refresh the app or mine another block.

Notes:
- Attempting to set the max amount when specifying spending or receiving channel balances will not always work yet. We will need to implement the calculation Blocktank uses to calculate channel costs to more accurately set these values. Right now it does its best to calculate what the user can and cannot afford.
